src/main/java/com/parasoft/demoapp/service/OrderService.java

```diff
@@ -44,6 +42,55 @@ public class OrderService {
     @Autowired
     private LocationService locationService;
 
+    @Transactional
+    public InventoryOperationRequestMessageDTO handleMessageFromResponseQueue(InventoryOperationResultMessageDTO operationResult) {
+        OrderEntity order = null;
+        String orderNumber = operationResult.getOrderNumber();
+        try {
+            order = getOrderByOrderNumber(orderNumber);
+            if(operationResult.getOperation() == InventoryOperation.DECREASE) {
+                switch (operationResult.getStatus()) {
+                    case SUCCESS:
+                        order = updateOrderStatus(orderNumber, OrderStatus.PROCESSED);
+                        OrderMQMessageDTO message =
+                                new OrderMQMessageDTO(orderNumber, order.getRequestedBy(), order.getStatus(), OrderMessages.THE_ORDER_IS_PROCESSED);
+                        orderMQService.sendToApprover(message);
+                        break;
+                    case FAIL:
+                        updateOrderStatus(orderNumber, OrderStatus.CANCELED);
+                        break;
+                    default:
+                        log.error(operationResult.getStatus() + " status is not supported");
+                }
+            }
+            return null;
+        } catch (OrderNotFoundException e) {
+            // TODO will be handled in separate task(rollback the inventory)
+            log.error("Order Not Found Exception:", e);
+            return new InventoryOperationRequestMessageDTO(InventoryOperation.INCREASE,
+                    operationResult.getOrderNumber(),
+                    null,
```

Comment:
@yshi-parasoft I think we need to discuss about theis situation when order is not found. In this situaton we need to increase the inventory but we can't give the items which need to be increased, because we have no inventoryInfos from response message and we can't get inventoryInfos from DB(order is not found).